lua/nabla.lua

```diff
@@ -333,27 +336,29 @@ local function gen_drawing(lines)
 end
 
 local function popup(overrides)
-  local buf = vim.api.nvim_get_current_buf()
-  local row, col = unpack(vim.api.nvim_win_get_cursor(0))
-  local line
-
-  if not row then
-    row, col = unpack(vim.api.nvim_win_get_cursor(0))
-  end
+    if not utils.in_mathzone() then
+        return
+    end
+    local line
 
-  local srow, scol, erow, ecol, del = find_latex_at(buf, row, col)
+    cur_line = line
 
-  if srow then
-    local lines = vim.api.nvim_buf_get_lines(buf, srow-1, erow, true)
-    lines[#lines] = lines[#lines]:sub(1, ecol)
-    lines[1] = lines[1]:sub(scol+1)
-    line = table.concat(lines, " ")
+    local math_node = utils.in_mathzone()
 
-  else
-    vim.api.nvim_echo({{"Please put the cursor inside an inline latex expression and try calling this function again.", "ErrorMsg"}}, false, {})
-    return
-  end
+    local srow, scol, erow, ecol = ts_utils.get_node_range(math_node)
+    local lines = vim.api.nvim_buf_get_text(0, srow, scol, erow, ecol, {})
 
+    line = table.concat(lines, " ")
+    line = line:gsub("%$", "")
+    line = line:gsub("\\%[", "")
+    line = line:gsub("\\%]", "")
+    line = line:gsub("^\\%(", "")
+    line = line:gsub("\\%)$", "")
+    local whitespace = string.match(line, "^(%s*)%S")
+    line = vim.trim(line)
```

Comment:
Does it also support \begin{equation} ... \end{equation} ?

Reply:
yes
because I just return the `text` node in the utils function
so the begin and end things aren't even there in the first place